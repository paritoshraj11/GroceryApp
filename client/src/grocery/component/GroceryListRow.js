import React from "react";
import { listRowStyle } from "../style";
import { dislikeIcon, likeIcon } from "../images";
import { isEqualObject } from "../clientUtility";

class ListRow extends React.Component {
  shouldComponentUpdate(nextProps) {
    return !isEqualObject(this.props, nextProps);
  }
  render() {
    let { data, likeAction, dislikeAction } = this.props;
    let { nameTextStyle, containerStyle, iconStyle, votesTextStyle } = listRowStyle || {};
    return (
      <li style={{ ...containerStyle }}>
        <span style={{ ...nameTextStyle }}>{data.name}</span>
        <span style={{ ...votesTextStyle }}>{data.votes}</span>
        <img style={{ ...iconStyle }} src={likeIcon} onClick={() => likeAction(data)} />
        <img style={{ ...iconStyle }} src={dislikeIcon} onClick={() => dislikeAction(data)} />
      </li>
    );
  }
}

export default ListRow;
