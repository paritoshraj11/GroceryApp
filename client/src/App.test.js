import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";
import GroceryListRow from "./grocery/component/GroceryListRow";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("list row test case", () => {
  const component = renderer.create(
    <GroceryListRow
      data={{
        _id: 1,
        name: "a",
        votes: 0
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
