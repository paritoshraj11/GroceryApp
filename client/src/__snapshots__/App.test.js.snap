// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list row test case 1`] = `
<li
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "gray",
      "borderBottomStyle": "solid",
      "borderBottomWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 2,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 10,
    }
  }
>
  <span
    style={
      Object {
        "flex": 1,
      }
    }
  >
    a
  </span>
  <span
    style={
      Object {
        "fontSize": 14,
        "paddingLeft": 5,
        "paddingRight": 5,
        "textAlign": "center",
      }
    }
  >
    0
  </span>
  <img
    onClick={[Function]}
    src="/images/like.png"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  />
  <img
    onClick={[Function]}
    src="/images/dislike.png"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  />
</li>
`;
